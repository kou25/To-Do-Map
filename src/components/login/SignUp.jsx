import React, { Component } from 'react'


export default class SignUp extends Component {
    state={
            email:'',
            password:'',
            firstName:'',
            lastName:''
        }
    handleChange=(e)=>{
        this.setState({
            [e.target.id]: e.target.value
        })
    }
    handleSubmit=(e)=>{
        e.preventDefault()
        console.log(this.state)
        this.props.history.push('/signin');
    }

    render() {
        return (
            <div className="container">
                <div data-testid="heading" className="d-sm-flex align-items-center justify-content-between mb-4">
                    <h1 style={{color:"#8B4513",  fontFamily:"cursive", margin:"30px 30%"}}>Sign Up<i className="fa fa-user-plus" aria-hidden="true" style={{marginLeft:"20px", color:"#8B4513"}}></i></h1>
                </div>
                <form onSubmit={this.handleSubmit} style={{width:"60%", marginLeft:"10%", fontFamily:"cursive"}}>
                <div className="form-group">
                    <label htmlFor="exampleInputEmail1">Email address</label>
                    <input type="email" 
                    className="form-control" 
                    placeholder="Enter email"
                    id="email"
                    required
                    onChange={this.handleChange}/>
                </div>
                <div className="form-group">
                    <label htmlFor="firstName">First Name</label>
                    <input type="text" 
                    className="form-control" 
                     placeholder="first name"
                     id="firstName"
                     required
                     onChange={this.handleChange}/>
                </div>
                <div className="form-group">
                    <label htmlFor="lastName">Last Name</label>
                    <input type="text" 
                    className="form-control" 
                     placeholder="last name"
                     id="lastName"
                     required
                     onChange={this.handleChange}/>
                </div>
                <div className="form-group">
                    <label htmlFor="exampleInputPassword1">Password</label>
                    <input type="password" 
                    className="form-control" 
                     placeholder="Password"
                     id="password"
                     required
                     onChange={this.handleChange}/>
                </div>
                <button data-testid="button" type="submit" className="btn btn-warning shadow mb-2"
                >Sign Up</button>
                </form>
            </div>
        )
    }
}
