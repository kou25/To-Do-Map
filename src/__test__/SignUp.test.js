import React from 'react';
import ReactDom from 'react-dom'
import SignUp from '../components/login/SignUp'
import {render, cleanup} from '@testing-library/react'
import "@testing-library/jest-dom/extend-expect"
import renderer from 'react-test-renderer'

afterEach(cleanup);
it("render without crashing",()=>{
  const div= document.createElement("div");
  ReactDom.render(
  <SignUp/>,div);
})

it("renders heading correctly", ()=>{
    const{getByTestId}=render(
    <SignUp/>)
    expect(getByTestId('heading')).toHaveTextContent('Sign Up')
  })

 

  it("renders button correctly", ()=>{
    const{getByTestId}=render(
    <SignUp/>)
    expect(getByTestId('button')).toHaveTextContent('Sign Up')
  })

  it('matches snapshot', ()=>{
    const tree=renderer.create(
    <SignUp/>).toJSON()
    expect(tree).toMatchSnapshot()
  })