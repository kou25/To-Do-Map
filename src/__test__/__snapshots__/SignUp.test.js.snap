// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="container"
>
  <div
    className="base-container"
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "marginTop": "90px",
      }
    }
  >
    <div
      className="card shadow"
      style={
        Object {
          "borderRadius": "20px",
          "height": "40rem",
          "width": "30rem",
        }
      }
    >
      <div
        className="content"
      >
        <div
          className="image"
          style={
            Object {
              "marginLeft": "25%",
              "marginTop": "40px",
            }
          }
        >
          <img
            alt="logo"
            src="signup.png"
          />
        </div>
        <form
          onSubmit={[Function]}
          style={
            Object {
              "fontFamily": "cursive",
              "marginLeft": "20%",
              "marginTop": "20px",
              "width": "60%",
            }
          }
        >
          <div
            className="form-group"
          >
            <label
              htmlFor="exampleInputEmail1"
            >
              Email address
            </label>
            <input
              className="form-control"
              id="email"
              onChange={[Function]}
              placeholder="Enter email"
              required={true}
              type="email"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="exampleInputName"
            >
              Name
            </label>
            <input
              className="form-control"
              id="name"
              onChange={[Function]}
              placeholder="Enter name"
              required={true}
              type="name"
              value=""
            />
          </div>
          <div
            className="form-group mt-4 mb-5"
          >
            <label
              htmlFor="exampleInputPassword1"
            >
              Password
            </label>
            <input
              className="form-control"
              id="password"
              onChange={[Function]}
              placeholder="Password"
              required={true}
              type="password"
              value=""
            />
          </div>
          <button
            className="btn btn-success  shadow"
            data-testid="button"
            type="submit"
          >
            Sign Up
          </button>
        </form>
      </div>
      <div
        className="right-side"
      >
        <div
          className="inner-container"
        >
          <div
            className="text"
          >
            Sign Up
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
